Bin/Expandflexentities.php

```diff
@@ -0,0 +1,197 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS AND CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Hoa\Devtools\Bin;
+
+use Hoa\Consistency\Consistency;
+use Hoa\Console;
+use Hoa\File;
+
+/**
+ * Class \Hoa\Devtools\Bin\ExpandFlexEntities
+ *
+ * This command resolve the class_alias function in destination to IDE.
+ *
+ * @copyright  Copyright © 2007-2017 Hoa community
+ * @license    New BSD License
+ */
+class Expandflexentities extends Console\Dispatcher\Kit
+{
+    /**
+     * Options description.
+     *
+     * @var array
+     */
+    protected $options = [
+        ['output', Console\GetOption::REQUIRED_ARGUMENT, 'o'],
+        ['dry-run', Console\GetOption::NO_ARGUMENT, 'd'],
+        ['verbose', Console\GetOption::NO_ARGUMENT, 'v'],
+        ['help', Console\GetOption::NO_ARGUMENT, 'h'],
+        ['help', Console\GetOption::NO_ARGUMENT, '?']
+    ];
+
+    /**
+     * The entry method.
+     *
+     * @return  void
+     */
+    public function main()
+    {
+        $dryRun  = false;
+        $output  = null;
+        $verbose = false;
+
+        while (false !== $c = $this->getOption($v)) {
+            switch ($c) {
+                case 'd':
+                    $dryRun = true;
+
+                    break;
+
+                case 'o':
+                    $output = $v;
+
+                    break;
+
+                case 'V':
+                    $verbose = true;
+
+                    break;
+
+                case 'h':
+                case '?':
+                    return $this->usage();
+
+                case '__ambiguous':
+                    $this->resolveOptionAmbiguity($v);
+
+                    break;
+            }
+        }
+
+        if (null === $output) {
+            return $this->usage();
+        }
+
+        $hoaPath = 'hoa://Library/';
+        $aliases = [];
+        $finder  = new File\Finder();
+        $finder
+            ->in(resolve($hoaPath, true, true))
+            ->name('#\.php$#')
+            ->files();
+
+        foreach ($finder as $file) {
+            $pathName = $file->getPathName();
+            $raw      = file_get_contents($pathName);
+
+            if (!preg_match('#flexEntity\(\'([^\']+)#', $raw, $class)) {
+                preg_match('#\nclass_alias\(\'([^\']+)#', $raw, $class);
+            }
+
+            if (empty($class)) {
+                continue;
+            }
+
+            $FQCN      = $class[1]; // Fully-Qualified Class Name
```

Comment:
You can store result in naming variable instead of using index you need to add in your regex the variable name

`preg_match('#\nclass_alias\(\'(?P<classname>[^\']+)#'`